includes/api/class-wc-rest-shipping-zone-methods-controller.php

```diff
@@ -103,6 +126,195 @@ public function get_items( $request ) {
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Request|WP_Error
+	 */
+	public function create_item( $request ) {
+		global $wpdb;
+
+		$method_id = $request['method_id'];
+		$zone      = $this->get_zone( $request['zone_id'] );
+		if ( is_wp_error( $zone ) ) {
+			return $zone;
+		}
+
+		$instance_id     = 0;
+		$wc_shipping     = WC_Shipping::instance();
+		$allowed_classes = $wc_shipping->get_shipping_method_class_names();
+		$count           = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$wpdb->prefix}woocommerce_shipping_zone_methods WHERE zone_id = %d", $zone->get_id() ) );
+		if ( in_array( $method_id, array_keys( $allowed_classes ) ) ) {
+			$wpdb->insert(
+				$wpdb->prefix . 'woocommerce_shipping_zone_methods',
+				array(
+					'method_id'    => $method_id ,
+					'zone_id'      => $zone->get_id(),
+					'method_order' => ( $count + 1 )
+				),
+				array(
+					'%s',
+					'%d',
+					'%d'
+				)
+			);
+			$instance_id = $wpdb->insert_id;
+		}
+
+		if ( $instance_id ) {
+			do_action( 'woocommerce_shipping_zone_method_added', $instance_id, $method_id, $zone->get_id() );
+		}
+
+		WC_Cache_Helper::get_transient_version( 'shipping', true );
+
+		$methods     = $zone->get_shipping_methods();
+		$method      = false;
+
+		foreach ( $methods as $method_obj ) {
+			if ( $instance_id === $method_obj->instance_id ) {
+				$method = $method_obj;
+				break;
+			}
+		}
+
+		if ( false === $method ) {
+			return new WP_Error( 'woocommerce_rest_shipping_zone_not_created', __( "Resource cannot be created.", 'woocommerce' ), array( 'status' => 500 ) );
+		}
+
+		$method = $this->update_fields( $instance_id, $method, $request );
+
+		$data = $this->prepare_item_for_response( $method, $request );
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Delete a shipping method instance.
+	 *
+	 * @param WP_REST_Request $request Full details about the request
+	 * @return WP_Error|boolean
+	 */
+	public function delete_item( $request ) {
+		global $wpdb;
+
+		$zone = $this->get_zone( $request['zone_id'] );
+		if ( is_wp_error( $zone ) ) {
+			return $zone;
+		}
+
+		$instance_id = (int) $request['instance_id'];
+		$force       = $request['force'];
+		$methods     = $zone->get_shipping_methods();
+		$method      = false;
+
+		foreach ( $methods as $method_obj ) {
+			if ( $instance_id === $method_obj->instance_id ) {
+				$method = $method_obj;
+				break;
+			}
+		}
+
+		if ( false === $method ) {
+			return new WP_Error( 'woocommerce_rest_shipping_zone_method_invalid', __( "Resource doesn't exist.", 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		$method = $this->update_fields( $instance_id, $method, $request );
+		$request->set_param( 'context', 'view' );
+		$response = $this->prepare_item_for_response( $method, $request );
+
+		if ( $force ) {
+			$wpdb->delete( $wpdb->prefix . 'woocommerce_shipping_zone_methods', array( 'instance_id' => $instance_id ) );
```

Comment:
Zone's `delete` method just deletes the whole zone. I'll add a `delete_shipping_method` to `WC_Shipping_Zone` though.